src/storage/src/list.cairo

```diff
@@ -17,119 +17,301 @@ struct List<T> {
 }
 
 trait ListTrait<T> {
+    /// Instantiates a new List with the given base address.
+    ///
+    ///
+    /// # Arguments
+    ///
+    /// * `address_domain` - The domain of the address. Only address_domain 0 is
+    /// currently supported, in the future it will enable access to address
+    /// spaces with different data availability
+    /// * `base` - The base address of the List. This corresponds to the
+    /// location in storage of the List's first element.
+    ///
+    /// # Returns
+    ///
+    /// A new List.
+    fn new(address_domain: u32, base: StorageBaseAddress) -> List<T>;
```

Comment:
Was wondering if it wouldn't make sense to have `base: felt252` as I imagine most if not all of the usage will be sth like `List::new(0, selector!("my_var_name"))` - no faffing about with `storage_base_address_from_felt252` needed.

Reply:
I'd rather keep it StorageBaseAddress as you can simply do `storage_var.address()`

Reply:
If you have `storage_var`, why is there a need for `new` then? Wouldn't that imply

```rust
struct Storage {
    storage_var: List<T>
}
```
?

Reply:
Given that the expected type for the Store trait is StorageBaseAddress is would expect the user to provide it as well to these functions. felts shouldn't really be exposed